src/rebar_prv_update.erl

```diff
@@ -168,9 +168,9 @@ update_deps_list(Pkg, PkgVsn, Deps, HexRegistry, State) ->
                                 cmpl(Dep1, rm_ws(Vsn), HexRegistry, State,
                                      DepsListAcc, fun ec_semver:lt/2);
                             {_, <<"==", Vsn/binary>>} ->
-                                [{Dep, Vsn} | DepsListAcc];
+                                [{AppName, {pkg, Dep, Vsn, undefined}} | DepsListAcc];
```

Comment:
yeah, they would be the hash. When no hash is specified (i.e. no lock file is found), the hash is replaced by `undefined` and updated as a package is fetched.

We don't *need* to do this, but I think this aligns the internal structure for pkg deps.

As mentioned in the PR message, I'm not quite sure how we can deal with the hashes for expected packages without looking at the data and lock files and may need to look at it later and repopulate the fields somehow. Currently the check is done on the first deps fetched (top-level) which may or may not be enough to cover all bases